launchable/commands/inspect/tests.py

```diff
@@ -1,23 +1,154 @@
+                "duration_min": round(self._results.filter_by_status("SUCCESS").total_duration_min(), 2)
+            },
+            "failure": {
+                "report_count": self._results.filter_by_status("FAILURE").total_count(),
+                "duration_min": round(self._results.filter_by_status("FAILURE").total_duration_min(), 2)
+            },
+            "skip": {
+                "report_count": self._results.filter_by_status("SKIPPED").total_count(),
+                "duration_min": round(self._results.filter_by_status("SKIPPED").total_duration_min(), 2)
+            }
+        }
+        result_json["results"] = []
+        for result in self._results.list():
+            result_json["results"].append({
+                "test_path": result._test_path,
+                "duration_sec": result._duration_sec,
+                "status": result._status,
+                "created_at": result._created_at
+            })
+
+        click.echo(json.dumps(result_json, indent=2))
+
+
+class TestResultTableDisplay(TestResultAbstractDisplay):
+    def __init__(self, results: TestResults):
+        super().__init__(results)
+
+    def display(self):
+        header = ["Test Path",
+                  "Duration (sec)", "Status", "Uploaded At"]
+        rows = []
+        for result in self._results.list():
+            rows.append(
+                [
+                    result._test_path,
+                    result._duration_sec,
+                    result._status,
+                    result._created_at,
+                ]
+            )
+        click.echo(tabulate(rows, header, tablefmt="github", floatfmt=".2f"))
+
+        summary_header = ["Summary", "Report Count", "Total Duration (min)"]
+        summary_rows = [
+            ["Total", self._results.total_count(),
+             self._results.total_duration_min()],
+            ["Success", self._results.filter_by_status("SUCCESS").total_count(),
+             self._results.filter_by_status("SUCCESS").total_duration_min()],
+            ["Failure", self._results.filter_by_status("FAILURE").total_count(),
+             self._results.filter_by_status("FAILURE").total_duration_min()],
+            ["Skip", self._results.filter_by_status("SKIPPED").total_count(),
+             self._results.filter_by_status("SKIPPED").total_duration_min()]]
+
+        click.echo(tabulate(summary_rows, summary_header, tablefmt="grid", floatfmt=["", ".0f", ".2f"]))
 
 
 @click.command()
 @click.option(
     '--test-session-id',
     'test_session_id',
     help='test session id',
-    required=True
+)
+@click.option(
+    '--json',
+    'is_json_format',
+    help='display JSON format',
+    is_flag=True
 )
 @click.pass_context
-def tests(context: click.core.Context, test_session_id: int):
+def tests(context: click.core.Context, test_session_id: int, is_json_format: bool):
+    if (test_session_id is None):
+        try:
+            session = require_session(None)
```

Comment:
I'm afraid that we raise an Exception here.

Reply:
I wrapped try/except around these logics, are there any problems?

Reply:
Oh sorry. the location is wrong. I mean https://github.com/launchableinc/cli/pull/801/files#diff-138881ab59bf9f6d072cd99143ee66eb54a5c07984619c5543ddd4a6f40970f0R152.

Reply:
Ah, I understood,  we shouldn't exit with status 0, right?
I'll fix it

Reply:
Yes, it is.

Reply:
Thank you.

Reply:
Than's nice catch